ostd/src/bus/pci/common_device.rs

```diff
@@ -96,19 +95,14 @@ impl PciCommonDevice {
 /// Base Address Registers manager.
 #[derive(Debug)]
 pub struct BarManager {
-    /// BARs, the bool indicate whether this bar should exposed to unprivileged part.
-    bars: [Option<(Bar, bool)>; 6],
+    /// There are at most 6 BARs in PCI device.
+    bars: [Option<Bar>; 6],
 }
 
 impl BarManager {
-    /// Gain access to the BAR space and return None if that BAR is set to be invisible or absent.
+    /// Gain access to the BAR space and return None if that BAR is absent.
     pub fn bar(&self, idx: u8) -> Option<Bar> {
-        let (bar, visible) = self.bars[idx as usize].clone()?;
-        if visible {
-            Some(bar)
-        } else {
-            None
-        }
+        self.bars[idx as usize].clone()
```

Comment:
Change the signature of this method to the following form.

```rust
    pub fn bar(&self, idx: u8) -> &Option<Bar> {
        &self.bar[idx as usize]
    }
```

This allows the caller to decide if a call to `clone` is really necessary.